lib/std/debug.zig

```diff
@@ -691,7 +691,7 @@ pub const StackIterator = struct {
         } else if (@hasDecl(os.system, "msync") and native_os != .wasi) {
             os.msync(aligned_memory, os.MSF.ASYNC) catch |err| {
                 switch (err) {
-                    os.MSyncError.UnmappedMemory => {
+                    os.MSyncError.HitRegionImmutable,os.MSyncError.UnmappedMemory => {
```

Comment:
Its not clear to me that this function (`isValidMemory`) should return false if `msync()` fails due to an immutable region.

Its hard to know what "valid" means without any comments, but it seems the callers are mostly just interested in "safe-to-read" memory, in which case immutable memory should be "valid".

Reply:
That's true, i've never thought about that, because it will be a bit strange to perform a write instruction into a immutable memory.

Reply:
> but it seems the callers are mostly just interested in "safe-to-read" memory,
False, because as i said before msync(2), do a async/sync write change on memory so it's not really interested on "safe-to-read" memory. the calls MFS_ASYNC conclude to it `The [msync](https://man.openbsd.org/msync.2#msync)() system call writes all pages with shared modifications`